### PR TITLE
[server] Fix GitLab testEmptyProject test

### LINKED ISSUE #4773: [server] `testEmptyProject` of the GitLab context parser fails
The `testEmptyProject` of the GitLab context parser fails:

https://github.com/gitpod-io/gitpod/blob/bc6c9f72c0d0a5a1085e5d37d67c01f1a5af4c31/components/server/src/gitlab/gitlab-context-parser.spec.ts#L531-L548

```
1) TestGitlabContextParser
       testEmptyProject:
     AssertionError: expected { Object (isFile, path, ...) } to have deep property 'ref'
      at TestGitlabContextParser.<anonymous> (src/gitlab/gitlab-context-parser.spec.ts:549:36)
      at Generator.next (<anonymous>)
      at fulfilled (src/gitlab/gitlab-context-parser.spec.ts:19:58)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

The actual result is:
```json
{
  "isFile": false,
  "path": "",
  "title": "gp-test-group/gp-test-empty-project - main",
  "revision": "",
  "repository": {
    "host": "gitlab.com",
    "name": "gp-test-empty-project",
    "owner": "gp-test-group",
    "cloneUrl": "https://gitlab.com/gp-test-group/gp-test-empty-project.git",
    "defaultBranch": "main",
    "private": false
  }
}
```

Opening the empty workspace actually works, though. It's just the test that fails and needs to be fixed.

### PR DESCRIPTION
**Pull request by corneliusludmann**
Fixes #4773

### How to test:

1. [Create a GitLab token](https://gitlab.com/-/profile/personal_access_tokens) (`api`, `read_user`, `read_repository`)
2. `cd components/server`
3. `export GITPOD_TEST_TOKEN_GITLAB='{"value": "YOUR-TOKEN"}'`
4. `yarn test`
5. Verify result:
```
[...]

  69 passing (34s)
  59 pending

Done in 41.21s.
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4865#" title="Author self-approved">corneliusludmann</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [corneliusludmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/gitlab/gitlab-context-parser.spec.ts
Comment (AlexTugarev) [inline components/server/src/gitlab/gitlab-context-parser.spec.ts:541]: this is somehow independent, but I believe they broke the API if that's really true now for uninitialized repos. how can an empty repo has a branch?

**Review by AlexTugarev — APPROVED** [review]
LGTM! Thanks for fixing!

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 13cfc6a0958961fc93f2800a1eaf8c6510c8cb4c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4865#pullrequestreview-709462902" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4865#" title="Author self-approved">corneliusludmann</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev,corneliusludmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
